src/game/game_state.c

```diff
@@ -72,6 +72,101 @@ typedef struct {
     int playback_id;
 } playing_sound;
 
+int game_state_get_assertion_operand(rec_assertion_operand *op, game_state *gs) {
+    if(op->is_literal) {
+        return op->value.literal;
+    } else {
+        object *obj =
+            game_state_find_object(gs, game_player_get_har_obj_id(game_state_get_player(gs, op->value.attr.har_id)));
+        har *har = object_get_userdata(obj);
+        switch(op->value.attr.attribute) {
+            case ATTR_X_POS:
+                return obj->pos.x;
+            case ATTR_Y_POS:
+                return obj->pos.y;
+            case ATTR_X_VEL:
+                return obj->vel.x;
+            case ATTR_Y_VEL:
+                return obj->vel.y;
+            case ATTR_STATE_ID:
+                return har->state;
+            case ATTR_ANIMATION_ID:
+                return obj->cur_animation->id;
+            case ATTR_HEALTH:
+                return har->health;
+            case ATTR_STAMINA:
+                return har->endurance;
+            case ATTR_OPPONENT_DISTANCE: {
+                object *obj_opp = game_state_find_object(
+                    gs, game_player_get_har_obj_id(game_state_get_player(gs, abs(op->value.attr.har_id - 1))));
+                return fabsf(obj->pos.x - obj_opp->pos.x);
+            }
+            default:
+                abort();
+        }
+    }
+}
+
+int game_state_check_assertion(rec_assertion *ass, game_state *gs) {
```

Comment:
this function should probably return a `bool`, and have a name that implies the meaning of true/false.

Reply:
I changed it to `game_state_assertion_is_met`